qemu/tests/check_coredump.py

```diff
@@ -1,198 +1,163 @@
+    return None
+
+
+def get_cores(resultsdir):
+    """
+    Recursive scan results dir to find core file
+    """
+    for root, dirs, files in os.walk(resultsdir):
+        for cfile in files:
+            if "core" in cfile:
+                filename = os.path.join(root, cfile)
+                yield filename
+        for sdir in dirs:
+            sdir = os.path.join(root, sdir)
+            get_cores(sdir)
 
-    line_break = "\n"
-    table_header = "Coredump file exists in the images:"
-    lines = []
-    sublines = []
-
-    for image, chk_msg in results:
-        lines += [image]
-
-        if isinstance(chk_msg, list):
-            for fname, ftime in chk_msg:
-                sublines += [fname + ", " + ftime]
 
-        lines += ["\t" + subline for subline in sublines]
+def scan_images_dir(imagesdir, resultsdir):
+    """
+    Scan image files under image dir and mount image to look
+    kernel dump files in each image.
+    """
+    dumpinfos = []
+    os_dep.command("guestfish")
+    for image in get_images(imagesdir, resultsdir):
+        dumpfiles = get_dumps(image, resultsdir)
+        if dumpfiles:
+            dumpinfos.append((image, dumpfiles))
+    return dumpinfos
 
-    all_lines = [""] + [table_header] + lines + [""]
 
-    return line_break.join(all_lines)
+def scan_results_dir(resultsdir):
+    """
+    Scan each test results directory to find coredump file
+    """
+    coreinfos = []
+    for corefile in get_cores(resultsdir):
+        dirname = os.path.dirname(os.path.dirname(corefile))
+        shortname = os.path.basename(shortname)
+        corefile = os.path.basename(corefile)
+        coreinfos.append((shortname, corefile))
+    return coreinfos
 
 
 @error.context_aware
 def run(test, params, env):
     """
-    We find out that sometimes the guest crashed between two
-    cases in a loop.  For example:
-    1. case A executed the steps and finished with good.
-    2. the post process do some check, pause or shutdown
-       operates.
-    3. guest crashed and reboot or just quit(But the case
-       already finished with good).
-    4. case B start and also didn't get the guest crashed
-       status.
-
-    Check if there is any core dump file in guest image .
-
-    1) Check all the existing guest images in the image directory.
-    2) Mount guest image on the host.
-    3) Check "C:\windows\dump" for Windows and core file for Linux.
-    4) If yes, copy them to working directory.
+    Post test, this case will scan results directory to detect
+    core dump file. And check each image file under image
+    directory to obtain guest kernel dumps. In the end, will go
+    through host dmesg log to report host error.
     """
-
-    # Preliminary
-    # yum install libguestfs libguestfs-tools libguestfs-winsupport
+    infos = ""
+    error.context("Scan results dirs to find core file", logging.info)
+    msg = []
+    try:
+        coreinfos = scan_results_dir(test.resultsdir)
+        if coreinfos:
```

Comment:
seems the resultsdir is incorrect.
the return of test.resultsdir is qemu.check_coredump/results
but we need to scan the core folder under the test case result
e.g 
results/default/virt.qemu.smp_1.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.boot

I create core.kvm.111 folder and file under boot, the result is pass